AllJoyn/Platform/Bridge/BridgeNET/BridgeNET.cpp

```diff
@@ -67,22 +67,71 @@ namespace BridgeNet
         std::shared_ptr<IBridge> bridge = IBridge::Get();
         bridge->AddAdapter(scriptAdapter);
         bridge->Initialize();
-    
+
         return S_OK;
     }
 
-    void BridgeJs::AddDevice(DeviceInfo^ info, 
-        String^ baseTypeXml, 
-        String^ script, 
+    // Wraps a shared_ptr to an IAdapterIORequest; preserves shared_ptr lifetime management
+    // of the request object and allows waiting for the request to complete.
+    ref class AdapterRequestAwaiter
+    {
+    public:
+        AdapterRequestAwaiter(const std::shared_ptr<IAdapterAsyncRequest>& request)
```

Comment:
Generally, you should pass a std::shared_ptr<...> as a copy, not by reference. In some limited/controlled cases it is OK, but in general it's a bad idea. Remove the "const" ..."&" and pass by value instead. This creates a copy which produces the correct refcounting behavior.